code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/URLUtils.java

```diff
@@ -0,0 +1,655 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.identity;
+
+import com.adobe.marketing.mobile.LoggingMode;
+import com.adobe.marketing.mobile.MobileCore;
+import java.io.UnsupportedEncodingException;
+
+public class URLUtils {
```

Comment:
This class should be package protected.
Also, any chance to use [URLEncoder.encode](https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html#encode(java.lang.String,%20java.lang.String)) instead of a custom class for your use-case?

Reply:
Let's remove public.